.github/workflows/push_snap.yml

```diff
@@ -0,0 +1,50 @@
+name: "Update Snap"
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - ".github/**"
+      - "!.github/workflows/push_aur.yml"
+      - "**.md"
+      - ".all-contributorsrc"
+      - ".gitignore"
+
+jobs:
+  snap:
+    name: "Update Snap"
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'cpeditor'
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+           submodules: recursive
+           ref: master
+           fetch-depth: 0
+
+      - name: Update CMake file
+        id: update-cmake-file
+        run: |
+          git fetch --tags
+          VERSION=$(git describe --long --tags --abbrev=8 | sed 's/-/./g' | cut -d '.' -f 1-4)
+          echo ::set-output name=VERSION::$VERSION
+          sed -i "s/project(cpeditor LANGUAGES CXX VERSION .*/project(cpeditor LANGUAGES CXX VERSION $VERSION)/g" CMakeLists.txt
```

Comment:
Why change this?

Reply:
Here `VERSION` should be a valid semver. Using the display version doesn't compile.

Reply:
Major.Minor.Patch.revisionNumber this is accepted in snap store and required to distinguish that new update is arrived. 

The AUR format does not works.

Reply:
Since each push to master will release to snap edge channel, I thought to use revision number as identifier.

Reply:
I think what you need is to change https://github.com/cpeditor/cpeditor/blob/9a349ffcbfbf0aafbfae3a49a8d1adb34da85868/cmake/cpeditor.appdata.xml.in#L29, right? Then you don't need to change `VERSION`. Use `DISPLAY_VERSION` if it's acceptable, or a new variable (`SNAP_VERSION`) if `DISPLAY_VERSION` doesn't work.

Reply:
> Using the display version doesn't compile.

Yes, using a sem version compiles. the `VERSION` is a sem version of format `major.minor.patch.tweak` tweak is revision since last tag.

Reply:
OK, it does compile. My concern is that using the `VERSION` differently from elsewhere could lead to potential bugs. We'd better use the same variable in the same way.

Reply:
> OK, it does compile. My concern is that using the `VERSION` differently from elsewhere could lead to potential bugs. We'd better use the same variable in the same way.

I will try to add a different variable `SNAP_VERSION` of above semver format and use it to configure appdata XML.